code/modules/mob/living/simple_animal/hostile/grue.dm

```diff
@@ -0,0 +1,591 @@
+			healths.icon_state = "health4"
+		else if (health > 0)
+			healths.icon_state = "health5"
+		else
+			healths.icon_state = "health6"
+//darkness level indicator
+		if (dark_dim_light==0)
+			healths2.icon_state= "lightlevel_dark"
+			healths2.name="nice and dark"
+		else if (dark_dim_light==1)
+			healths2.icon_state= "lightlevel_dim"
+			healths2.name="adequately dim"
+		else if (dark_dim_light==2)
+			healths2.icon_state= "lightlevel_bright"
+			healths2.name="painfully bright"
+
+/mob/living/simple_animal/hostile/grue/Life()
+	..()
+
+	//process shadow power and health according to current tile brightness level
+	if (stat!=DEAD)
+		if(isturf(loc))
+			var/turf/T = loc
+			current_brightness=10*T.get_lumcount()
+		else												//else, there's considered to be no light
+			current_brightness=0
+//		visible_message("<span class='warning'>\The [src] is in brightness level [current_brightness] with [health] health and [shadowpower] shadowpower.</span>") //debug
+		if(current_brightness<=bright_limit_gain&&!ismoulting) //moulting temporarily stops healing via darkness
+			dark_dim_light=0
+			apply_damage(-1*burnmalus*regenbonus*hg_mult*(bright_limit_gain-current_brightness),BURN) //boost juveniles and adults heal rates a bit also using burnmalus
+		else if(current_brightness>bright_limit_drain) 														//lose health in light
+			dark_dim_light=2
+
+			to_chat(src, "<span class='warning'>The bright light scalds you!</span>")
+			playsound(src, 'sound/effects/flesh_squelch.ogg', 50, 1)
+			apply_damage(burnmalus*hd_mult*(current_brightness-bright_limit_drain),BURN)								//scale light damage by lifestage**(1/3) to avoid juveniles and adults from becoming too tanky to light
+		else
+			dark_dim_light=1
+		if(current_brightness<=bright_limit_gain&&!ismoulting)
+			shadowpower = min(maxshadowpower,shadowpower+pg_mult*(bright_limit_gain-current_brightness))	   //gain power in dark
+		else if(current_brightness>bright_limit_drain)
+			shadowpower = max(0,shadowpower-pd_mult*(current_brightness-bright_limit_drain))				  //drain power in light
+
+		//process speed modifiers
+		if(dark_dim_light==0)
+			speed=base_speed*speed_m_dark
+		else if(dark_dim_light==1)
+			speed=base_speed*speed_m_dim
+		else if(dark_dim_light==2)
+			speed=base_speed*speed_m_light
+
+		if(ismoulting)
+			moulttimer--
+			if(moulttimer<=0)
+				complete_moult()
+
+	regular_hud_updates()
+	standard_damage_overlay_updates()
+
+/mob/living/simple_animal/hostile/grue/New()
+	..()
+	add_language(LANGUAGE_GRUE)
+	default_language = all_languages[LANGUAGE_GRUE]
+	init_language = default_language
+//	if(thislifestage)
+//		lifestage=thislifestage
+//	else
+//		lifestage=3 //default to adult
+	lifestage_updates() //update the grue's sprite and stats according to the current lifestage
+
+/mob/living/simple_animal/hostile/grue/proc/lifestage_updates() //Initialize or update lifestage-dependent stats
+	var/tempHealth=health/maxHealth
+	if(lifestage==1)
+		name = "grue larva"
```

Comment:
maybe make this mob subtypes

Reply:
The reason I did it like this is to make it easier to switch between life stages as the grue moults and such. Do you mean it would be better to spawn a new mob, transfer the mind in, and then delete the original mob?